Modules/_asynciomodule.c

```diff
@@ -2269,14 +2272,32 @@ Returns the remaining number of cancellation requests.
 static PyObject *
 _asyncio_Task_uncancel_impl(TaskObj *self)
 /*[clinic end generated code: output=58184d236a817d3c input=68f81a4b90b46be2]*/
-/*[clinic end generated code]*/
 {
     if (self->task_num_cancels_requested > 0) {
         self->task_num_cancels_requested -= 1;
     }
     return PyLong_FromLong(self->task_num_cancels_requested);
 }
 
+/*[clinic input]
+_asyncio.Task._check_future
+
+    future: object
+
+Return False if task and future loops are not compatible.
+[clinic start generated code]*/
+
+static PyObject *
+_asyncio_Task__check_future_impl(TaskObj *self, PyObject *future)
+/*[clinic end generated code: output=2473f5b55d524bd5 input=76b4fa671a14b681]*/
+{
+    int res = task_check_future_exact(self, future);
+    if (res < 0) {
+        return NULL;
+    }
+    return PyBool_FromLong(res);
+}
```

Comment:
It looks like the function can be simplified by using a `bool` return converter. It cannot be expressed in a simple diff so I [created a branch from your one](https://github.com/python/cpython/compare/asyncio-task-check...arhadthedev:_check_future-bool?expand=1).